phylogeny/prototypeSelection.py

```diff
@@ -448,3 +449,83 @@ def prototype_selection_constructive_protoclass(dm: DistanceMatrix,
         prototypes.append(np.asarray(minVals).argmin())
 
     return [dm.ids[idx] for idx in prototypes]
+
+
+def prototype_selection_destructive_maxdist(dm, num_prototypes):
+    '''Heuristically select k prototypes for given distance matrix.
+
+       Prototype selection is NP-hard. This is an implementation of a greedy
+       correctness heuristic: Start with the complete set and interatively
+       remove elements until the number of required prototypes is left.
+       The decision which element shall be removed is based on the minimal
+       distance sum this element has to all other.
+
+    Parameters
+    ----------
+    dm: skbio.stats.distance.DistanceMatrix
+        Pairwise distances for all elements in the full set S.
+    num_prototypes: int
+        Number of prototypes to select for distance matrix.
+        Must be >= 2, since a single prototype is useless.
+        Must be smaller than the number of elements in the distance matrix,
+        otherwise no reduction is necessary.
+
+    Returns
+    -------
+    list of str
+        A sequence holding selected prototypes, i.e. a sub-set of the
```

Comment:
From the return, it looks like this is a subset of the IDs of the matrix not the elements?